packages/fluentui/react-bindings/src/hooks/useAccessibility.ts

```diff
@@ -19,44 +19,41 @@ type UseAccessibilityResult = (<SlotProps extends Record<string, any>>(
   unstable_wrapWithFocusZone: (children: React.ReactElement) => React.ReactElement;
 };
 
-type MergedProps<SlotProps extends Record<string, any>> = SlotProps &
-  Partial<AccessibilityAttributesBySlot> & {
-    onKeyDown?: (e: React.KeyboardEvent, ...args: any[]) => void;
-  };
+type UserProps = {
+  onKeyDown?: KeyboardEventHandler;
+};
 
-const mergeProps = <SlotProps extends Record<string, any>>(
-  slotName: string,
-  slotProps: SlotProps,
-  definition: ReactAccessibilityBehavior
-): MergedProps<SlotProps> => {
-  const finalProps: MergedProps<SlotProps> = {
-    ...definition.attributes[slotName],
-    ...slotProps
-  };
-  const slotHandlers = definition.keyHandlers[slotName];
+type MergedProps<SlotProps extends Record<string, any> = any> = SlotProps & Partial<AccessibilityAttributesBySlot> & UserProps;
 
-  if (slotHandlers) {
-    const onKeyDown = (e: React.KeyboardEvent, ...args: any[]) => {
-      if (slotHandlers && slotHandlers.onKeyDown) {
-        slotHandlers.onKeyDown(e);
-      }
+const useAccessibility = <Props>(behavior: Accessibility<Props>, options: UseAccessibilityOptions<Props> = {}) => {
+  const { actionHandlers, debugName = 'Undefined', mapPropsToBehavior = () => ({}), rtl = false } = options;
 
-      if (slotProps.onKeyDown) {
-        slotProps.onKeyDown(e, ...args);
-      }
-    };
+  const definition = getAccessibility(debugName, behavior, mapPropsToBehavior(), rtl, actionHandlers);
 
-    finalProps.onKeyDown = onKeyDown;
-  }
+  const slotHandlers = React.useRef<Record<string, KeyboardEventHandler>>({});
+  const slotProps = React.useRef<Record<string, UserProps>>({});
 
-  return finalProps;
-};
+  const getA11Props: UseAccessibilityResult = (slotName, userProps) => {
+    slotProps.current[slotName] = userProps;
 
-const useAccessibility = <Props>(behavior: Accessibility<Props>, options: UseAccessibilityOptions<Props> = {}) => {
-  const { actionHandlers, debugName = 'Undefined', mapPropsToBehavior = () => ({}), rtl = false } = options;
-  const definition = getAccessibility(debugName, behavior, mapPropsToBehavior(), rtl, actionHandlers);
+    if (!slotHandlers.current[slotName]) {
+      slotHandlers.current[slotName] = (e, ...args) => {
+        const accessibilityHandler = definition.keyHandlers[slotName].onKeyDown;
```

Comment:
`keyHandlers` for a slot can be undefined. That was previously checked on line 38 but you don't have the check in the new code.

Also the previous contract was that the onKeyDown listener was added to an element only if it was handling anything. And that could change based on `props` change. Now the listener is added and never removed.
I think it would be better to be unstable in these edge cases and add/remove the listener when needed.

You can use `ButtonBehavior` to test both issues.
- It does not add the listener by default -> first issue.
- If you rerender with `as='div'`, it is supposed to add the listener, if you remove the`as` later, it should remove the listener -> second issue.